app/components/animated_number/index.tsx

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useMemo} from 'react';
+import {Animated, Easing, type LayoutChangeEvent, type StyleProp, Text, type TextStyle, View} from 'react-native';
+
+interface Props {
+    animateToNumber: number;
+    fontStyle?: StyleProp<TextStyle>;
+    animationDuration?: number;
+    easing?: ((input: number) => number) | undefined;
+  }
+
+const NUMBERS = Array(10).fill(null).map((_, i) => i);
+
+const usePrevious = (value: number) => {
+    const ref = React.useRef<number|undefined>();
+    React.useEffect(() => {
+        ref.current = value;
+    });
+
+    if (typeof ref.current === 'undefined') {
+        return 0;
+    }
+
+    return ref.current;
+};
+
+const AnimatedNumber = ({
+    animateToNumber,
+    animationDuration,
+    fontStyle,
+    easing,
+}: Props) => {
+    const prevNumber = usePrevious(animateToNumber);
+    const animateToNumberString = String(Math.abs(animateToNumber));
+    const prevNumberString = String(Math.abs(prevNumber));
+
+    const animateToNumbersArr = Array.from(animateToNumberString, Number);
```

Comment:
0/5 but maybe this should be called something to do with "digits"? It took me a bit to understand why we convert the number to a string and then break that into an array